DM-20924: Clean ups to support PyPI distribution

timj opened the pull request.

timj committed:
Move corrections files into module

This makes installation easier as package_data.

timj committed:
Install corrections files and translators/serializers

timj committed:
Use pkg_resources to obtain local corrections files

timj committed:
Allow zipped egg for installed package

timj committed:
Only ignore top level bin directory

timj committed:
Move translate_header implementation into module code

timj committed:
Move almost all translate_header logic into testable function

timj committed:
Add ability to specify output streams for translateheader functions

This significantly simplifies testing.

timj committed:
Add tests for translate_header script functionality

timj committed:
Add bin module to setuptools

timj committed:
Need to use a normal module for "bin"

timj committed:
Include sphinx docs for translate_header command

jonathansick commented on setup.cfg:
```diff
@@ -18,6 +18,8 @@ package_dir=
     =python
 packages=
     astro_metadata_translator
+    astro_metadata_translator.translators
```
This is fine, but I'm interested for my own sake why this was needed. With my work I'm always doing `setuptools.find_packages` in a `setup.py` file so it must automatically be adding these sub packages for me. Automatic package detection doesn't work here I guess?

timj replied:
I do have `options.packages.find` set later but it really was not installing these files at all until I explicitly mentioned them here. Maybe I'm doing it wrong?  Sometimes the migration from `setup.py` to `setup.cfg` is non-obvious but I am trying to follow the modern best practice here and define everything in the cfg file.

timj replied:
Thank you for the prod. It made me look again and I am now using `find_packages` using the completely non-obvious syntax found in `setup.cfg`: see 4c5b6163cbac00abc7a7f84930888cefbc4ea95a

jonathansick commented on doc/index.rst:
```diff
@@ -56,3 +56,15 @@ Python API reference
 
 .. automodapi:: astro_metadata_translator
+
+Command Line Utilities
```
In the long run you might want to move this documentation out from the midst of the Python API references. Perhaps the solution is to drop the API Reference for `astro_metadata_translator.bin.translateheader` since it's unlikely that API wouldn't be needed. Or you might want to move the CLI reference pages to another rst file and link to with with a toctree? Just a thought.

timj replied:
Given how little docs there are I think in the short term I will move this above the API section.

timj committed:
Do a full install for travis

A developer install does not fully test the setup.py because
these checks were passing even when the setup.cfg was not
installing corrections files or the translators. The dev
install points back to the developer directory so is
ignoring the failure to install.  A full install will
install an egg and use the egg for subsequent
testing.

jonathansick reviewed (approved):
This looks great. Nice work on the testing.

I always personally package data issue to be hard to debug until I make a release candidate build on PyPI and verify that everything got packaged right, but I can't see anything wrong with this.

timj committed:
Use find_packages to locate python packages

I was missing the

packages=find:

option...

timj committed:
Switch translate_header.py to use entry points with setup.py

timj committed:
Return bad exit status if no files could be translated

This updates the `main()` API to match that expected by
standard Python entry points.

timj commented:
@jonathansick I just switched the translate_header.py command to use entry points. It was trivial now that I had refactored the command following your advice. Thanks for that.

timj committed:
Read long description from README file

timj committed:
Fix author name

For now there is a single author so less confusing to say that.

jonathansick commented:
This looks great :+1:

timj committed:
Fix the Jira component URL

timj committed:
Add a change log
